pkg/mimir/mimir.go

```diff
@@ -626,6 +630,29 @@ func InheritCommonFlagValues(log log.Logger, fs *flag.FlagSet, common CommonConf
 				return fmt.Errorf("can't inherit common flags for %q: %w", desc, err)
 			}
 		}
+		if common.ClusterValidationLabel == "" {
+			// Nothing to inherit because origin was not set.
+			continue
+		}
+		for _, grpcClientConfig := range inheritance.ClusterValidationLabel {
+			if grpcClientConfig == nil {
+				continue
+			}
+			if grpcClientConfig.ClusterValidationLabel != "" {
+				// Can't inherit because destination was set.
+				continue
+			}
+			if common.ClusterValidationLabel == grpcClientConfig.ClusterValidationLabel {
+				// Already the same, no need to touch.
+				continue
+			}
+			level.Debug(log).Log(
+				"msg", "Inheriting flag value",
+				"flag_name", "cluster_value_label", "origin_value", common.ClusterValidationLabel,
```

Comment:
The `flag_name` isn't the correct one here, and also, I would try to match the existing log line from `inheritFlags`, otherwise the logs just don't look consistent, it's a nit, but we can do better.